test/autorest/generated/lrogroup/pollers.go

```diff
@@ -107,6 +107,12 @@ func (p *productPoller) Poll(ctx context.Context) (*http.Response, error) {
 }
 
 func (p *productPoller) FinalResponse(ctx context.Context) (*ProductResponse, error) {
+	// checking if there was a FinalStateVia configuration to re-route the final GET
+	// request to the value specified in the FinalStateVia property on the poller
+	err := p.pt.setFinalState()
```

Comment:
I think you can move the call to `setFinalState()` to `createPollingTracker()` and `resumePollingTracker()` so you don't have to call it from the generated code.  It would also let you remove the method from the interface.

Reply:
So I thought about that when developing this. However, I think that could lead to some problems, since the final polling url could technically be modified from a subsequent polling request. Therefore, I thought it was better to have it in the FinalResponse to override those changes and consolidate this functionality in one place. What do you think?

Reply:
Good point, let's leave it as-is.